contracts/src/factory/HyperdriveRegistry.sol

```diff
@@ -46,17 +62,184 @@ contract HyperdriveRegistry is
 
     /// @inheritdoc IHyperdriveGovernedRegistry
     function setHyperdriveInfo(
```

Comment:
Very much personal preference but I kinda like the explicitness of separate add/update/remove as opposed to combining them

Reply:
I hear you, but this will be more efficient for the use-cases that I'm imagining. Let's say that we need to make a bulk update to the registry (which is something that I anticipate happening in the future). We may need to add new entries, update others, and delete yet others. In this case, being able to call `setHyperdriveInfo` with the whole batch would be cheaper since you'd only need to call one function in one transaction. A related point is that the events are more consolidated. The only event you need to accumulate over to have an up-to-date state of the repo is `HyperdriveInfoUpdated`.